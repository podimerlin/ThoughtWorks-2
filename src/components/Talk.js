import React, { Component } from 'react'

class Talk extends Component {
    constructor(props) {
        super(props);
        var initDate = new Date();
        initDate.setHours(9,0,0,0);
        this.state = { ...props, time: initDate };                
    }

    // add minutes to 9am and display a 12 hour time string
    changeStringToTime(time, mins) {
      mins = typeof this.props.length === 'undefined' ? mins : mins - this.props.length;
      time = new Date(time.getTime() + mins*60000);

      return (
        time.toLocaleString('en-US', { hour: 'numeric',minute:'numeric', hour12: true })
      );
    }

    render() {
        return(
            <div> {this.changeStringToTime(this.state.time, this.props.spentTime)} {this.state.title} {this.state.lengthtext} </div>
        )
    }   
}

export default Talk;