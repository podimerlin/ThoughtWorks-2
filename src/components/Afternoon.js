import React, { Component } from 'react';
import Talk from './Talk';

class Afternoon extends Component {
constructor(props) {
    super(props);

    this.spendTime =  240; //min 420 max 480
  }


  render() {
    this.spendTime = 240;    
    if (this.props.talks.length === 0) {
      return (
        <div>
          No Afternoon Talks Today
        </div>
      )
    }
    return( 
      <div className="afternoon-sess session">
      {
        this.props.talks.map(talk => {
          this.spendTime += talk.props.length;

          return (
            <Talk 
              key={talk.props.id}
              title={talk.props.title}
              lengthtext={talk.props.lengthtext}
              length={talk.props.length}
              spentTime={this.spendTime}
            />
          );
        }, this)
      }
      <Talk 
        className="networking"
        key={-1}
        title={'Networking Event'}
        spentTime={this.spendTime}
      />
      </div>

    );
  }


}

export default Afternoon;