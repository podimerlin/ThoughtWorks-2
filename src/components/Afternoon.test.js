import React from 'react';
import { shallow } from 'enzyme';
import Afternoon from './Afternoon';
import Talk from './Talk';

describe('Afternoon', () => {
  const props = {talks: [<Talk key={1} id={1} title={'title1'} length={10} lengthtext={'10min'} />]};
  const afternoon = shallow(<Afternoon {...props}/>);
  
  it('renders correctly', () => {
      expect(afternoon).toMatchSnapshot();
  });

  it('creates Talk Component', () => {
    expect(afternoon.find('Talk').exists()).toBe(true);
  })

  it('create Networking Talk Component', () => {
    expect(afternoon.find('Talk.networking').exists()).toBe(true);
  })

})