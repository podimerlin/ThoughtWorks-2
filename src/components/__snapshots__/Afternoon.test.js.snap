// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Afternoon renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Afternoon
    talks={
        Array [
            <Talk
              id={1}
              length={10}
              lengthtext="10min"
              title="title1"
        />,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Talk
            length={10}
            lengthtext="10min"
            spentTime={250}
            title="title1"
/>,
        ],
        <Talk
          className="networking"
          spentTime={250}
          title="Networking Event"
/>,
      ],
      "className": "afternoon-sess session",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "length": 10,
          "lengthtext": "10min",
          "spentTime": 250,
          "title": "title1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "-1",
        "nodeType": "class",
        "props": Object {
          "className": "networking",
          "spentTime": 250,
          "title": "Networking Event",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Talk
              length={10}
              lengthtext="10min"
              spentTime={250}
              title="title1"
/>,
          ],
          <Talk
            className="networking"
            spentTime={250}
            title="Networking Event"
/>,
        ],
        "className": "afternoon-sess session",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "length": 10,
            "lengthtext": "10min",
            "spentTime": 250,
            "title": "title1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "-1",
          "nodeType": "class",
          "props": Object {
            "className": "networking",
            "spentTime": 250,
            "title": "Networking Event",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
